docs/source/model/dependencies.rst

```diff
@@ -0,0 +1,453 @@
+
+Testing online inference endpoint with a virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you want to test your model by actually running the online inference server, you can use MLflow `serve` API.
+This will create a virtual environment with your model and dependencies, similarly to the `predict` API, but will start the inference server
+and expose the REST endpoints. Then you can send a test request and validate the response.
+Please refer to `CLI reference <../cli.html#mlflow-models>`_ for more detailed usage for the `serve` API.
+
+.. code-block:: bash
+
+    mlflow models serve -m runs:/<run_id>/model -p <port>
+    # In another terminal
+    curl -X POST -H "Content-Type: application/json" \
+        --data '{"inputs": [[1, 2], [3, 4]]}' \
+        http://localhost:<port>/invocations
+
+While this is a reliable way to test your model before deployment, one caveat is that the virtual environment doesn't absorb the OS-level differences
+between your machine and the production environment. For example, if you are using MacOS as a local dev machine but your deployment target is
+running on Linux, you may encounter some issues that are not reproducible in the virtual environment.
+
+In this case, you can use a Docker container to test your model. While it doesn't provide full OS-level isolation unlike virtual machines e.g. we
+can't run Windows containers on Linux machines, Docker covers some popular test scenarios such as running different versions of Linux or simulating
+Linux environment on Mac/Windows.
+
+Testing online inference endpoint with a Docker container
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+MLflow `build-docker` API for CLI and Python, which builds an Ubuntu-based Docker image for serving your model.
+The image will contain your model and dependencies and has an entrypoint to start the inference server. Similarly to the `serve` API,
+you can send a test request and validate the response.
+Please refer to `CLI reference <../cli.html#mlflow-models>`_ for more detailed usage for the `build-docker` API.
+
+.. code-block:: bash
+
+    mlflow models build-docker -m runs:/<run_id>/model -n <image_name>
+    docker run -p <port>:8080 <image_name>
+    # In another terminal
+    curl -X POST -H "Content-Type: application/json" \
+        --data '{"inputs": [[1, 2], [3, 4]]}' \
+        http://localhost:<port>/invocations
+
+
+.. _model-dependencies-troubleshooting:
+
+Troubleshooting
+---------------
+
+.. _how-to-fix-dependency-errors-in-model:
+
+How to fix dependency errors when serving my model?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+One common issue during model deployment is dependency issue. When logging/saving your model, MLflow tries to infer the
+model dependencies and save them as part of the MLflow Model metadata. However, this inference is not always accurate and may
```

Comment:
I'd say that the implication here might be a bit off the mark. The inference is accurate in the sense that we capture the correct versions of packages. The issue can be that the set of requirements is not **complete** due to how package dependencies can be defined within certain libraries (local imports, [extras] requirements that aren't inferred properly for completeness. 
I would adjust this a bit to avoid saying that it's not 'accurate', but instead state that it's potentially not 'complete'.

Reply:
Totally agree, my brain started to type "accurate/cy" when seeing word "inference":p